src/clarity-angular/datagrid/render/header-renderer.ts

```diff
@@ -47,16 +63,32 @@ export class DatagridHeaderRenderer implements OnDestroy {
     }
 
     public computeWidth(): number {
-        if (this.strictWidth) {
-            // We do NOT set the width here, since we know the user already provided it.
-            this.renderer.setElementClass(this.el.nativeElement, STRICT_WIDTH_CLASS, true);
-            return this.strictWidth;
-        } else {
-            let width = this.domAdapter.scrollWidth(this.el.nativeElement);
-            this.renderer.setElementClass(this.el.nativeElement, STRICT_WIDTH_CLASS, false);
+
+        let width: number = this.strictWidth;
+
+        this.renderer.setElementClass(this.el.nativeElement, STRICT_WIDTH_CLASS, !!width);
+
+        if (this.columnResizer.columnResizeBy) {
+            this.renderer.setElementStyle(this.el.nativeElement, "width", width + "px");
+            this.columnResizer.columnResizeBy = 0;
+            this.widthSet = false;
+        }
+
+        if (!width) {
+
+            width = this.domAdapter.scrollWidth(this.el.nativeElement);
+
+            // a column shouldn't shrink beyond its minWidth when other columns get larger by resizing
+            if (width < this.minWidth) {
+                width = this.minWidth;
+            }
```

Comment:
Why do we have to handle this manually? The way we do it right now, if you set a min-width on the header, the `scrollWidth` computed takes it into account already.

I actually have an issue with this header renderer having to worry about the resizer's min width. See my previous comment about putting the default min-width of 96px in the CSS, it would mean this renderer would never have to know anything about the resizer.